tests/basic/pool_function_test.go

```diff
@@ -1338,3 +1339,71 @@ var _ = Describe("{CheckPoolLabelsAfterResizeDisk}", func() {
 
 //})
 
+
+
+var _ = Describe("{PoolVolUpdateResizeDisk}", func() {
+
+	var testrailID = 34542876
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/tests/view/34542876
+
+	BeforeEach(func() {
+		StartTorpedoTest("PoolVolUpdateResizeDisk", "Increase the HA replica of the volume and expand pool using resize-disk during the increase", nil, testrailID)
+		contexts = scheduleApps()
+	})
+	JustBeforeEach(func() {
+		poolIDToResize = pickPoolToResize()
+		log.Infof("Picked pool %s to resize", poolIDToResize)
+		poolToResize = getStoragePool(poolIDToResize)
+		storageNode, err = GetNodeWithGivenPoolID(poolIDToResize)
+		log.FailOnError(err, "Failed to get node with given pool ID")
+
+	})
+	JustAfterEach(func() {
+		AfterEachTest(contexts)
+	})
+
+	AfterEach(func() {
+		appsValidateAndDestroy(contexts)
+		EndTorpedoTest()
+	})
+	It("Increase the HA replica of the volume and expand pool using resize-disk during the increase", func() {
+		var newRep int64
+		var currRep int64
+		volSelected, err := GetVolumeWithMinimumSize(contexts, 10)
+		log.FailOnError(err, "error identifying volume")
+		opts := volume.Options{
+			ValidateReplicationUpdateTimeout: replicationUpdateTimeout,
+		}
+		Step("Increase the HA replica of the volume", func() {
+			currRep, err = Inst().V.GetReplicationFactor(volSelected)
+			log.FailOnError(err, fmt.Sprintf("err getting repl factor for  vol : %s", volSelected.Name))
+			newRep = currRep
+			// If the HA is 3, reduce it by 1, so that we can increase it later
+			if currRep == 3 {
+				newRep = currRep - 1
+				err = Inst().V.SetReplicationFactor(volSelected, newRep, nil, nil, true, opts)
+				log.FailOnError(err, fmt.Sprintf("err setting repl factor  to %d for  vol : %s", newRep, volSelected.Name))
```

Comment:
Do we need to wait and verify the replication factor was changed before proceeding?

Reply:
No. The test states to do pool expansion while the replica increase is going on.

Reply:
fixed.